File: leetcode/python/1422_maximum_score_after_splitting_string.py
class Solution:
    def maxScore(self, s: str) -> int:
        left = 0
        right = s.count('1')
        res = 0
        for c in s[:-1]:
            if c == '0':
                left += 1
            else:
                right -= 1
            res = max(res, left + right)
        return res

File: leetcode/rust/src/1422_maximum_score_after_splitting_string.rs
impl Solution {
    pub fn max_score(s: String) -> i32 {
        let mut left = 0;
        let mut right = s.chars().filter(|&c| c == '1').count();
        let mut res = 0;

        for c in s.chars().take(s.len() - 1) {
            if c == '0' {
                left += 1;
            } else {
                right -= 1;
            }
            res = res.max((left + right) as i32);
        }
        res
    }
}